docs/recipes/WritingTests.md

```diff
@@ -130,16 +132,22 @@ function mockStore(getState, expectedActions, done) {
       },
 
       dispatch(action) {
+        if (hasDoneBefore) {
+          return
+        }
+      
         const expectedAction = expectedActions.shift()
 
         try {
           expect(action).toEqual(expectedAction)
           if (done && !expectedActions.length) {
             done()
+            hasDoneBefore = true
```

Comment:
I *think* removing this will solve your issues. There are a few different scenarios to account for:
* We got an expected action and there are still actions left. Continue as normal.
* We got an expected action and there are no actions left.
  * Call `done()`
  * Do NOT no-op dispatch. We've made it through all of our expected actions and have a successful test (so far). Any future actions will fail the expect check, just as an invalid action would. See the next bullet
* We got an unexpected action (it doesn't matter if we have actions left, since we want to fail the test at this point)
  * Throw, call `done(e)`, and set the hasDoneBefore flag.

You will still end up calling `done` multiple times if all actions are successful and then you pass an additional action. However, mocha will report the error correctly despite done being called twice (since you're passing an error on the second `done` call). Subsequent actions will no-op since anytime you reach the catch block, no more actions will be processed.

I have not considered how other testing frameworks would handle this scenario.